### DISCUSSION TITLE
Changing int to float for KeyframeIntervalSec ?

### DISCUSSION THREAD

**Discussion by RobertMengerBMW** [obsproject/obs-studio#4603] [General]
Hi there, 

for some applications, a keyframe-interval of 1 second is too long, especially if high image quality and/or low latency is required.

Does anyone see any problems in changing the type of this value from int to float? 

kind regards
Robert

**Comment by matoi974** [comment]
If you're using NVENC or AMF, you can install StreamFX and use their custom implementation of keyint, where you can specify it in frames or as float with two decimal points. I don't think obs-studio will directly implement keyint in frames, that'd allow for user errors.
If you're stuck with x264, you can use custom FFmpeg output, where keyint may be overridden manually, though you won't receive much tech support with that setup. I think the correct keys would be g=30:keyint=30, but don't quote me on that, I might have forgotten.

**Reply by RobertMengerBMW** [reply]
could you point me in the direction of how to customize FFMPEG output, please?

**Reply by Fenrirthviti** [reply]
Please use the forums or Discord for support, we do not provide support over Discussions. I am going to close this thread as it is a support request.

https://obsproject.com/help
